browser/src/Editor/SimpleNeovimEditor.tsx

```diff
@@ -0,0 +1,120 @@
+/**
+ * NeovimEditor.ts
+ *
+ * IEditor implementation for Neovim
+ */
+
+import * as React from "react"
+
+import { IncrementalDeltaRegionTracker } from "./../DeltaRegionTracker"
+import { NeovimInstance } from "./../neovim"
+import { CanvasRenderer, DOMRenderer, INeovimRenderer } from "./../Renderer"
+import { NeovimScreen } from "./../Screen"
+
+import * as Config from "./../Config"
+
+// import { PluginManager } from "./../Plugins/PluginManager"
+
+import { Keyboard } from "./../Input/Keyboard"
+import { IEditor } from "./Editor"
+
+import { NeovimRenderer } from "./NeovimRenderer"
+
+export class DummyPluginManager {
+    public getAllRuntimePaths(): string[] {
+        return []
+    }
+
+    public startPlugins(neovimInstance: NeovimInstance): void {
+        // no-op
+    }
+}
+
+export class SimpleNeovimEditor implements IEditor {
+
+    private _neovimInstance: NeovimInstance
+    private _deltaRegionManager: IncrementalDeltaRegionTracker
+    private _renderer: INeovimRenderer
+    private _screen: NeovimScreen
+
+    private _pendingTimeout: any = null
+    private _pendingAnimationFrame: boolean = false
+
+    constructor(
+        private _config: Config.Config = Config.instance(),
+    ) {
+        // TODO: How to get rid of this?
+        this._neovimInstance = new NeovimInstance(new DummyPluginManager(), 100, 100)
+        this._neovimInstance.setInitVim("C:/oni/test.vim")
+        this._deltaRegionManager = new IncrementalDeltaRegionTracker()
+        this._screen = new NeovimScreen(this._deltaRegionManager)
+
+        this._renderer = this._config.getValue("editor.renderer") === "canvas" ? new CanvasRenderer() : new DOMRenderer()
+
+        this._render()
+
+        this._config.registerListener(() => this._onConfigChanged())
+
+        this._neovimInstance.on("action", (action: any) => {
+            this._renderer.onAction(action)
+            this._screen.dispatch(action)
+
+            this._scheduleRender()
+
+            if (!this._pendingTimeout) {
+                this._pendingTimeout = setTimeout(() => this._onUpdate(), 0)
+            }
+        })
+
+        const keyboard = new Keyboard()
```

Comment:
> I'm assuming when you say extract keyboard you mean have detached from an editor and whichever editor has focus will receive the events right?

That's exactly right!

> The difference would be instead of an interface have an abstract, and then have a static instance/private instance for _currentEditor, and that way you don't have to change who keyboard is sending to because it can just be _currentEditor.sendKeys() and then _currentEditor can process it based on it's overrided method but can also pull common code from AbstractEditor, thoughts?

Yes, I think that's a really good idea. It actually lines up a lot with what I've been thinking for an API. I'd like for plugins to know who the activeEditor is - something like:
`Oni.editors.activeEditor`

As well as knowing when the editor changes - something like:
`Oni.editors.onActiveEditorChanged((newEditor, oldEditor) => { ... })`

But I don't want the plugins to need to always be listening for active editor changed - I might just have a plugin that logs to the console when I change mode, like:
```
const activate = (Oni) => {
Oni.editors.activeEditor.onModeChanged((newMode) => console.log("New mode: " + newMode);
}
```

And I'd want to have that work across the currently active editor, regardless of which one it is. Sounds like a similiar case to input - I might want to send input to the active editor, and not care about the implementation.

So I think it's a great idea, thanks for sharing! I'd picture refactoring this a bit and have a common core editor class as you mention (with some of the shared functionality)